SourceCode/include/vec.hpp

```diff
@@ -1,20 +1,148 @@
 #pragma once
+#include <cstdint>
+#include <cmath>
+#include <type_traits>
 
-class Vec3
+// chaos math
+namespace cm {
+template<typename T, int N>
```

Comment:
The parametrization of the size of the vector could over complicate the code, because not all operations are defined in all dimensions. It's better to assume that all vectors are in 3D space.

Reply:
I am using a Vec2 for resolution and pixel indices. By templating I can overload all operators for all Vec sizes at once.
The operations that are only defined in specific dimensions can be implemented using the partial specialization of Vec. Those operations most of the time need a custom implementation anyway, as they are usually not implementable in a size-agnostic way.
It makes it really easy to use any Vec size and implement size specific operations if needed while making it indistinguishable from a non-templated version for the user.
So, I think I'd rather stick to this version, if it's okay :)